src/Domain/ElementId.hpp

```diff
@@ -60,7 +51,9 @@ class ElementId {
 
   ElementId<VolumeDim> id_of_parent(size_t dim) const noexcept;
 
-  constexpr size_t block_id() const noexcept { return block_id_; }
+  size_t block_id() const noexcept {
+    return segment_ids_[0].block_id();
```

Comment:
For discussion: I would feel more comfortable if there were an ASSERT that the different segment_ids agreed on the block id. Perhaps this is overkill, because they're all set together in the ElementId constructor.. but such an ASSERT could catch future bugs. What do you think?

Reply:
I have no objection to adding an ASSERT :) Better to have them and never need them than to have someone spend hours figuring out some stupid bug :)